packages/next/build/webpack/plugins/nextjs-require-cache-hot-reloader.ts

```diff
@@ -57,30 +54,9 @@ export class NextJsRequireCacheHotReloader implements WebpackPluginInstance {
   apply(compiler: Compiler) {
     compiler.hooks.assetEmitted.tap(
       PLUGIN_NAME,
-      (file, { targetPath, content }) => {
-        this.currentOutputPathsWebpack5.add(targetPath)
+      (_file, { targetPath, content }) => {
         deleteCache(targetPath)
         clearModuleContext(targetPath, content.toString('utf-8'))
-
-        if (
```

Comment:
Double checked this and it is calling `assetEmitted` when the file is written for `.__sc_client__` so I'm not sure why it wouldn' have that before but this check is no longer needed, especially with the added `compilation.entries()` check: https://github.com/vercel/next.js/pull/38633/files#diff-d57600652ac21cc49a0603a849016288549cd5bc7386dca47eb5eed4b9579459R75